src/main/java/filter/expression/Qualifier.java

```diff
@@ -136,23 +131,61 @@ public static boolean process(IModel model, FilterExpression expr, TurboIssue is
         return exprWithNormalQualifiers.isSatisfiedBy(model, issue, new MetaQualifierInfo(metaQualifiers));
     }
 
+    /**
+     * Get all milestones that can be aliased with the keyword "current".
+     */
+    private static List<TurboMilestone> getMilestonesWithAlias(IModel model, List<String> repoIds) {
+        List<TurboMilestone> milestones = new ArrayList<>();
+        // add milestones with due date first
+        milestones.addAll(model.getMilestones().stream()
+                .filter(ms -> ms.getDueDate().isPresent())
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .sorted((a, b) -> getMilestoneDueDateComparator().compare(a, b))
+                .collect(Collectors.toList()));
+
+        // Special case: if there is only one open milestone, it should be included regardless of whether
+        // it has due date or not.
+        // In this case, if it is not included already (the open milestone does not
+        // have due date), append it to the milestone list.
+        List<TurboMilestone> openMilestones = getOpenMilestones(model, repoIds);
+        if (openMilestones.size() == 1) {
+            TurboMilestone openMilestone = openMilestones.get(0);
+            boolean hasDueDate = openMilestone.getDueDate().isPresent();
+            if (!hasDueDate) {
+                milestones.add(openMilestone);
+            }
+        }
+
+        return milestones;
+    }
+
+    private static List<TurboMilestone> getOpenMilestones(IModel model, List<String> repoIds) {
+        return model.getMilestones().stream()
+                .filter(ms -> ms.isOpen())
+                .filter(ms -> repoIds.contains(ms.getRepoId().toLowerCase()))
+                .collect(Collectors.toList());
+    }
+
     private static Optional<Integer> getCurrentMilestoneIndex(List<TurboMilestone> allMilestones) {
```

Comment:
For the original one:

Input = milestones that can be aliased, sorted by due date
Output = index of `current` milestone within the given milestone list

Logic =

1. If there is no milestones that can be aliased, there is no `current` milestone hence return empty index.
2. Find first open milestone which is either:
  - overdue but still have open issues in it (if there is no open issues, assume the admin forgot to close the milestone), or
  - not overdue yet.
3. If no such open milestone, consider the last milestone in the list as the `current-1` (only open milestone can be current). In this case, return size of milestone list as `current` index (hence `current` returns no result, but `current-1)